src/util-base64.c

```diff
@@ -157,7 +157,8 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
     }
 
     if (bbidx > 0 && bbidx < 4 && ((!valid && mode == BASE64_MODE_RFC4648))) {
-        padding = B64_BLOCK - bbidx;
+        /* Decoded bytes for 1 or 2 base64 encoded bytes is 1 */
+        padding = bbidx > 1 ? B64_BLOCK - bbidx : 2;
```

Comment:
This commit lacks unit tests.
I think that for `bb_idx == 1`, padding needs to be `B64_BLOCK - bbidx` = 3, and thus no bytes can be decoded (we have only 6 bits)

Reply:
> (we have only 6 bits)

agreed. But, should we not have at least one decoded byte by making use of the padding? If I try any algorithms, even 1 byte does get decoded to something..
Please let me know what you think..

Reply:
I think generally we are greedy, and we want to decode as much as possible, esp is other implementations exist that do this as well.